lib/datadog/profiling/component.rb

```diff
@@ -207,28 +207,16 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
 
         return false unless heap_profiling_enabled
 
-        if RUBY_VERSION.start_with?("2.") && RUBY_VERSION < "2.7"
+        if RUBY_VERSION < "3.1"
           Datadog.logger.warn(
-            "Heap profiling currently relies on features introduced in Ruby 2.7 and will be forcibly disabled. " \
-            "Please upgrade to Ruby >= 2.7 in order to use this feature."
+            "Current Ruby version (#{RUBY_VERSION}) cannot support heap profiling due to VM bugs/limitations. " \
+            "Please upgrade to Ruby >= 3.1 in order to use this feature. Heap profiling has been disabled."
```

Comment:
Not a deal-breaker, but does it make sense to include the link to the bug which forced us to do it? I have a feeling that we may want to say just `VM limitations` to be nice.

And second question, if someone have patched ruby 2.7 and still want to use the profiler, they are able to do it only with version prior to new release, right?